Add OverrideMagneto flag

ziogaschr opened the pull request:
Resolves task on #365

ziogaschr committed:
cmd/geth,cmd/utils,core,eth,eth/ethconfig: replace ecbp1100Flag and OverrideBerlinFlag usage with OverrideMagnetoFlag

ziogaschr commented on cmd/geth/usage.go:
```diff
@@ -58,7 +58,6 @@ var AppHelpFlagGroups = []flags.FlagGroup{
 			utils.IdentityFlag,
 			utils.LightKDFFlag,
 			utils.WhitelistFlag,
-			utils.ECBP1100Flag,
```
@meowsbits I have also removed ECBP1100 flag as we already transitioned to it. I once thought to leave/keep it, as it's not really a hard-fork.

meowsbits replied:
Would it be possible to keep the ECBP1100 flag around, but not touching the genesis/persistent override stuff? 
I'm just thinking about edge cases where it might be really helpful to be able to disable it if needed with `--ecbp1100=9999999999`. Or even different `--artificialfinality` flag? Thoughts?

ziogaschr replied:
Oh, I forgot that all cases can be configured with `--artificialfinality`. I will bring the flag back, it makes total sense.

ziogaschr replied:
Resolved with https://github.com/etclabscore/core-geth/pull/367/commits/5cd22b53b2e549a6d15871bc2ca4a05a6297fd1f

meowsbits commented on core/genesis.go:
```diff
@@ -66,9 +66,21 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *genesisT.Genesis,
+			if err := genesis.SetEIP2565Transition(&n); err != nil {
+				return genesis, stored, err
+			}
+			if err := genesis.SetEIP2315Transition(&n); err != nil {
```
`EIP2315 Simple Subroutines` should be removed. There's a PR in doing this that hasn't been merged yet. https://github.com/ethereumclassic/ECIPs/pull/419/files.

meowsbits replied:
Done with https://github.com/etclabscore/core-geth/pull/367/commits/af67af42d0df129231eb81686689b47fc6530c3e

meowsbits commented on core/genesis.go:
```diff
@@ -108,11 +120,21 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *genesisT.Genesis,
+		if err := newcfg.SetEIP2565Transition(&n); err != nil {
+			return newcfg, stored, err
+		}
+		if err := newcfg.SetEIP2315Transition(&n); err != nil {
```
`EIP2315 Simple Subroutines` should be removed. There's a PR in doing this that hasn't been merged yet. https://github.com/ethereumclassic/ECIPs/pull/419/files.

meowsbits replied:
Done with https://github.com/etclabscore/core-geth/pull/367/commits/af67af42d0df129231eb81686689b47fc6530c3e

meowsbits committed:
core: remove EIP2315 from Magneto overrides

This EIP was removed at the last minute from
the Berlin spec, and is -- pending merge --
slated for removal at the Magneto spec as well.

https://github.com/ethereumclassic/ECIPs/pull/419

Date: 2021-04-30 07:05:01-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: fixup comments (go fmt)

Date: 2021-04-30 07:11:38-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
cmd/geth,cmd/utils,eth/ethconfig: bring back the ECBP1100 flag

Based on https://github.com/etclabscore/core-geth/pull/367#discussion_r623792593

ziogaschr committed:
eth: allow ecbp1100 activation block override
